src/electron/platform/android/rule-information-provider.ts

```diff
@@ -50,14 +50,26 @@ export class RuleInformationProvider {
 
     private getColorContrastUnifiedResolution = (ruleResultsData: RuleResultsData): UnifiedResolution => {
         const ratio = ruleResultsData.props['Color Contrast Ratio'] as string;
-        const foreground = (ruleResultsData.props['Foreground Color'] as string).substring(2);
-        const background = (ruleResultsData.props['Background Color'] as string).substring(2);
+
+        const foreground = this.getColorValue(ruleResultsData, 'Foreground Color');
+        const background = this.getColorValue(ruleResultsData, 'Background Color');
 
         return this.buildUnifiedResolution(
-            `The text element has insufficient contrast of ${ratio}. Foreground color: #${foreground}, background color: #${background}). Modify the text foreground and/or background colors to provide a contrast ratio of at least 4.5:1 for regular text, or 3:1 for large text (at least 18pt, or 14pt+bold).`,
+            `The text element has insufficient contrast of ${ratio}. Foreground color: ${foreground}, background color: ${background}). Modify the text foreground and/or background colors to provide a contrast ratio of at least 4.5:1 for regular text, or 3:1 for large text (at least 18pt, or 14pt+bold).`,
         );
     };
 
+    private getColorValue(ruleResultsData: RuleResultsData, propertyName: string): string {
+        let result = 'NO VALUE AVAILABLE';
+        const value = ruleResultsData.props[propertyName] as string;
+
+        if (value) {
+            result = `#${value.substring(2)}`;
```

Comment:
```value.substring(2)``` is a bit unhelpful to read. Why the use of 2 here? Might be simplified with a named variable.

Reply:
updated